src/main/java/com/aws/greengrass/mqtt/bridge/clients/LocalMqtt5Client.java

```diff
@@ -88,11 +92,18 @@ public class LocalMqtt5Client implements MessageClient<MqttMessage> {
     private static final long DEFAULT_TCP_MQTT_PORT = 1883;
     private static final long DEFAULT_SSL_MQTT_PORT = 8883;
 
-    private boolean clientStarted = false; // crt close is not idempotent
+    private final MQTTClientKeyStore.UpdateListener onKeyStoreUpdate = new MQTTClientKeyStore.UpdateListener() {
+        @Override
+        public void onCAUpdate() {
+            LOGGER.atInfo().log("New CA cert available, reconnecting client");
+            reset();
```

Comment:
instead of directly calling reset, how about just scheduling the reset? That way you don't block this callback thread (which CDA owns) and you can avoid multiple threads calling into reset

Reply:
same for the other callback too of course

Reply:
yeah that's a good point, will give it a go

Reply:
done